### PR TITLE
Enable Elastic Search initialization and temporalio docker image updates

### PR DESCRIPTION
**Pull request by markmark206**
**What changed?**
* `temporalio/server` image is now capable of performing auto-setup (if requested via `AUTO_SETUP=true` environment variable), adding `vim` to the image, for troubleshooting convenience,
* Adding `docker-images` target to Makefile,
* Initializing Elastic Search schema separately from other database schemas (to be used by helm chart deploys),
* Adding `admin-tools` to `docker-compose-es.yaml` 

**Why?**
* To simplify our docker images a bit, and to position us for retiring `temporalio/auto-setup`
* To make it easier to build and test docker images on a dev box,
* To make it easier to connect to a running instance of temporal via `admin-tools` container.

**How did you test it?**
1. I build docker images by running `make docker-images` on my machine,
2. I ran `ADMIN_TOOLS_TAG=markmark-local SERVER_TAG=markmark-local docker-compose -f docker-compose-es.yml up` and verified that services start. 
3. With the docker-compose instance above running, dropped into `admin-tools`, by running `docker exec -it docker_temporal-admin-tools_1 bash`, and tested sure workflows can be started and examined, via `tctl ... wf list | scan | count work`
4. deployed these changes with helm chart and performed test 3 in k8s cluster. 

**Potential risks**

Our docker images might be broken,  docker-compose- and helm-installed environments won't work. Hopefully, this will be caught by our test pipelines. If this happens, we just fix this or roll back the change.

### CODE REVIEW

**Comment by mfateev** [comment]
Can we address https://github.com/temporalio/temporal/issues/308 here?

**Comment by markmark206** [comment]
@mfateev i updated `docker-compose-es.yml` file to cap the memory used by ElasticSearch.

tl;dr: ES startup memory usage shrunk, 1.3G ->  384M. As a bonus, I can now run with only 2GB (which was the default on my mac) allocated to Docker, instead of having to raise it to 4GB to make things work).

If you think this addresses the concern in #308, I will also update and close that ticket.

----
Details:

Here is memory usage before and after the change.

Before (running on mac os, docker memory set to 4gb, because the default of 2gb caused ElasticSearch to immediately exit):
```
$ docker stats docker_elasticsearch_1 docker_kafka_1 docker_zookeeper_1 docker_cassandra_1
CONTAINER ID        NAME                     CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
2fcceff99f3f        docker_elasticsearch_1   0.34%               1.282GiB / 4.824GiB   26.57%              1.51MB / 1MB        132MB / 614kB       62
b3be9c9e8686        docker_kafka_1           1.58%               783.7MiB / 4.824GiB   15.86%              132MB / 203MB       107MB / 58.8MB      83
a37ff6df1912        docker_zookeeper_1       0.13%               62.97MiB / 4.824GiB   1.27%               4.7MB / 3.05MB      54.4MB / 848kB      27
4fae3f6984d6        docker_cassandra_1       1.09%               1.436GiB / 4.824GiB   29.77%              80.9MB / 94.7MB     126MB / 181MB       91
```
 
After the change (docker memory is now set to 2Gb, and things still startup without any issues):
```
CONTAINER ID        NAME                     CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
0cef8fc5ee86        docker_elasticsearch_1   0.77%               383.9MiB / 4.824GiB   7.77%               7.95kB / 5.84kB     0B / 664kB          56
CONTAINER ID        NAME                     CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
0cef8fc5ee86        docker_elasticsearch_1   0.77%               383.9MiB / 4.824GiB   7.77%               7.95kB / 5.84kB     0B / 664kB          56
CONTAINER ID        NAME                     CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
0cef8fc5ee86        docker_elasticsearch_1   0.99%               383.9MiB / 4.824GiB   7.77%               7.95kB / 5.84kB     0B / 664kB          56
```

I verified that tctl wf count / scan operations still work.
